Modules/_asynciomodule.c

```diff
@@ -441,8 +441,16 @@ future_schedule_callbacks(asyncio_state *state, FutureObj *fut)
         return 0;
     }
 
-    for (i = 0; i < len; i++) {
+    // Beware: An evil 'call_soon' could change fut_callbacks or its items
+    // (see https://github.com/python/cpython/issues/125789 for details).
+    for (i = 0;
+         fut->fut_callbacks != NULL && i < PyList_GET_SIZE(fut->fut_callbacks);
+         i++) {
         PyObject *cb_tup = PyList_GET_ITEM(fut->fut_callbacks, i);
+        if (!PyTuple_CheckExact(cb_tup) || PyTuple_GET_SIZE(cb_tup) != 2) {
+            PyErr_SetString(PyExc_RuntimeError, "corrupted callback tuple?");
```

Comment:
```suggestion
            PyErr_SetString(PyExc_RuntimeError, "corrupted callback tuple");
```

Is this a typo?

Reply:
Totally. I wanted to check something :')